lib/sqlalchemy/sql/functions.py

```diff
@@ -440,7 +487,11 @@ def __call__(self, *c, **kwargs):
             package = None
 
         if package is not None:
-            func = _registry[package].get(fname)
+            reg = _registry[package]
```

Comment:
**mike bayer** (zzzeek) wrote:

So here, this is a *very* common codepath and it occurs such that an extremely common outcome is that the function is not registered, and we will be creating a new Function object.

The next *very* common codepath is that someone is calling upon one of the existing generic functions like count() and there's only one of them registered.

So I want to reduce the lookups here for these codepaths; if we make use of the _CASE_SENSITIVE symbol in [2] above, I think this can become:


    func = _registry[package].get(fname.lower())
    if func is _CASE_SENSITIVE:
        func = _case_sensitive_registry[package]get(fname)

    if func is not None:
       return func(*c, **o)

That way we don't add any new lookups unless they are using the deprecated / unusual case sensitive path.  does that work?

Reply:
Indeed, with the _CASE_SENSITIVE symbol it is much better.